.github/workflows/CD.yml

```
@@ -1,78 +1,84 @@
 name: CD
 
 on:
+  workflow_dispatch:
   push:
     branches: [master]
 
 env:
   CARGO_TERM_COLOR: always
+
 jobs:
-  docker:
+  build:
     runs-on: ubuntu-latest
     steps:
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
-
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2.0.0
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Build and push
-        id: docker_build
-        uses: docker/build-push-action@v2
+        uses: docker/build-push-action@v3.1.0
         with:
           file: ../../docker/Dockerfile
           push: true
           tags: ghcr.io/gear-tech/node:latest
 
   deploy:
     runs-on: ubuntu-latest
-    needs: docker
+    needs: build
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: Read secrets
+        uses: hashicorp/vault-action@v2.4.1
+        with:
+          url: ${{ secrets.VAULT_URL }}
+          tlsSkipVerify: true
+          method: github
+          githubToken: ${{ secrets.VAULT_TOKEN }}
+          secrets: |
+            kv/ssh/keys nodes | SSH_KEY ;
+            kv/infra/testnet-nodes GEAR_NODE_1 | GEAR_NODE_1 ;
+            kv/infra/testnet-nodes GEAR_NODE_2 | GEAR_NODE_2 ;
+            kv/infra/testnet-nodes GEAR_NODE_3 | GEAR_NODE_3 ;
+            kv/infra/testnet-nodes GEAR_NODE_4 | GEAR_NODE_4 ;
+            kv/infra/testnet-nodes RPC_NODE | RPC_NODE ;
+            kv/infra/testnet-nodes ARCHIVE_NODE | ARCHIVE_NODE ;
+            kv/infra/testnet-nodes ARCHIVE_RPC_NODE | ARCHIVE_RPC_NODE ;
+            kv/infra/other LOKI_URL | LOKI_URL ;
+            kv/telegram_bots/alerts chat_id | TG_CHAT ;
+            kv/telegram_bots/alerts token | TG_TOKEN
 
-      - name: Install Info
-        run: |
-            ls -l .
-            ls -l ../..
-            pwd
+      - name: Checkout
+        uses: actions/checkout@v3.0.2
```

I don't see the reason to choose a concrete tag. Better to use a branch with the latest release within the major version. See for example: https://github.com/actions/checkout/
```suggestion
        uses: actions/checkout@v3
```